### PR TITLE
[idrac_certificates]: Fixing issue #737 for SSL CSR request fails

### LINKED ISSUE #737: [BUG]: iDrac SSL CSR request fails
### Bug Description

CSR generation with key length of 4096 takes longer than for 2048 which is expected, however, module returns failure instead of waiting for finish.
Changing timeout did not help. 
I think (guessing here) that the problem is that iDrac returns unexpected text (info about longer processing time) and module is not ready for that.

### Component or Module Name

dellemc.openmanage.idrac_certificates

### Ansible Version

Ansible 2.15.2

### Python Version

Python 3.11.8

### iDRAC/OME/OME-M version

Latest

### Operating System

RHEL 9

### Playbook Used

```yaml
- name: Generate HTTPS certificate signing request
      dellemc.openmanage.idrac_certificates:
        idrac_ip: "{{ inventory_hostname }}"
        idrac_user: "{{ idrac_user }}"
        idrac_password: "{{ idrac_user_pw }}"
        command: "generate_csr"
        certificate_type: "HTTPS"
        certificate_path: "path/to/certificate"
        cert_params: "{{ idrac_cert_params }}"
        timeout: 120
      register: _idrac_csr
      delegate_to: localhost
```

### Logs

{
  "msg": "HTTP Error 503: Service Unavailable",
  "error_info": {
    "error": {
      "@Message.ExtendedInfo": [
        {
          "Message": "The Generate CSR operation is taking longer duration than expected. Wait few minutes for the operation to complete and perform the Download CSR operation. For more information, see the iDRAC Redfish API Guide available on the support site.",
          "MessageArgs": [
            " "
          ],
          "MessageArgs@odata.count": 1,
          "MessageId": "IDRAC.2.8.SYS537",
          "RelatedProperties": [],
          "RelatedProperties@odata.count": 0,
          "Resolution": "No response action is required.",
          "Severity": "Informational"
        }
      ],
      "code": "Base.1.12.GeneralError",
      "message": "A general error has occurred. See ExtendedInfo for more information"
    }
  },

### Steps to Reproduce

Set iDrac to use 4096 RSA keys for CSR.

You can do that via command line `racadm set iDRAC.Security.CsrKeySize 4096`

Run Ansible task to request CSR (`dellemc.openmanage.idrac_certificates`).

### Expected Behavior

Generate CSR

### Actual Behavior

Return info about the fact that it is gonna take more time. CSR is actually generated, but it takes longer.

### Screenshots

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ABHISHEK-SINHA10**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: tests/unit/plugins/modules/test_idrac_certificates.py [hunk 1]
```diff
@@ -299,6 +299,31 @@ def test_build_generate_csr_payload(self, idrac_default_args):
         payload = self.module._build_generate_csr_payload(f_module, None)
         assert payload["AlternativeNames"] == ['192.198.2.1,192.198.2.2,X.X.X.X']
```
Comment (trisha-dell) [inline tests/unit/plugins/modules/test_idrac_certificates.py:300]: Are we okay to expose these IPs. Are they dummy?

Comment (trisha-dell) [reply tests/unit/plugins/modules/test_idrac_certificates.py:300]: Please change the copyright year. Applicable to all instances.

Comment (ABHISHEK-SINHA10) [reply tests/unit/plugins/modules/test_idrac_certificates.py:300]: yeah those are local IP address and dummy. sure will change the copyright

**Comment by ABHISHEK-SINHA10** [comment]
Test coverage

image

File: tests/unit/plugins/modules/test_idrac_certificates.py (hunk 1 above)
Comment (ABHISHEK-SINHA10) [reply tests/unit/plugins/modules/test_idrac_certificates.py:300]: @trisha-dell please re-check and resolve conversation.

**Review by Saksham-Nautiyal — DISMISSED** [review]

**Review by trisha-dell — APPROVED** [review]

**Review by trisha-dell — DISMISSED** [review]

**Comment by ABHISHEK-SINHA10** [comment]
Test coverage
image

**Review by rajshekarp87 — APPROVED** [review]

**Review by Kritika-Bhateja-03 — APPROVED** [review]

**Review by Saksham-Nautiyal — APPROVED** [review]

**Review by ShivamSh3 — APPROVED** [review]

**Pull request merged**